homeassistant/components/iss/__init__.py

```diff
@@ -1,18 +1,69 @@
 """The iss component."""
 from __future__ import annotations
 
+from datetime import datetime, timedelta
+import logging
+from typing import TypedDict
+
+import pyiss
+import requests
+from requests.exceptions import HTTPError
+
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
 
 from .const import DOMAIN
 
+_LOGGER = logging.getLogger(__name__)
+
 PLATFORMS = [Platform.BINARY_SENSOR]
 
 
+class IssData(TypedDict):
+    """Typed dict representation of data returned from pyiss."""
+
+    number_of_people_in_space: int
+    current_location: dict[str, str]
+    is_above: bool
+    next_rise: datetime
+
+
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up this integration using UI."""
     hass.data.setdefault(DOMAIN, {})
+    latitude = hass.config.latitude
+    longitude = hass.config.longitude
+
+    iss = pyiss.ISS()
+
+    async def async_update() -> IssData:
+        try:
+
+            def update(latitude: float, longitude: float) -> IssData:
```

Comment:
This function doesn't need to be nested within `async_update`. We could even move it to the module level if we pass `iss` as a parameter. Flat is better than nested.